rmgpy/pdep/sls.py

```diff
@@ -101,27 +101,210 @@ def solve_me_fcns(f, jac, M, p0, t):
+            x0[i] = 1.0
+            p0 = get_initial_condition(network,x0,indices)
+            out = etM @ p0
+            xsout = states_to_configurations(network,indices,out)
+            ddt = states_to_configurations(network,indices,np.dot(M,out))
+            return xsout,ddt
+
+        def run_equilibrium(network, channel):
+            i = (network.isomers+network.reactants+network.products).index(channel)
+            x0 = network.eq_ratios/np.sum(network.eq_ratios)
+            x0[i] = 0.0
+            p0 = get_initial_condition(network,x0,indices)
+            out = etM @ p0
+            xsout = states_to_configurations(network,indices,out)
+            ddt = states_to_configurations(network,indices,np.dot(M,out))
+            return xsout,ddt
+
+    elif method == "eigen":
+        def run_single_source(network, channel):
+            i = (network.isomers+network.reactants+network.products).index(channel)
+            x0 = np.zeros(len(network.isomers+network.reactants+network.products))
+            x0[i] = 1.0
+            p0 = get_initial_condition(network,x0,indices)
+            c0 = np.linalg.solve(eigvecs,p0)
+            cf = np.exp(eigvals*tau) * c0
+            out = eigvecs @ cf
+            xsout = states_to_configurations(network,indices,out)
+            ddt = states_to_configurations(network,indices,np.dot(M,out))
+            return xsout,ddt
+
+        def run_equilibrium(network, channel):
+            i = (network.isomers+network.reactants+network.products).index(channel)
+            x0 = network.eq_ratios/np.sum(network.eq_ratios)
+            x0[i] = 0.0
+            p0 = get_initial_condition(network,x0,indices)
+            c0 = np.linalg.solve(eigvecs,p0)
+            cf = np.exp(eigvals*tau) * c0
+            out = eigvecs @ cf
+            xsout = states_to_configurations(network,indices,out)
+            ddt = states_to_configurations(network,indices,np.dot(M,out))
+            return xsout,ddt
     else:
-        xs = np.zeros(network.n_isom+network.n_reac+network.n_prod)
-    for i in range(network.n_isom):
-        cum = np.float64(0.0)
-        for r in range(network.n_grains):
-            for s in range(network.n_j):
-                index = indices[i,r,s]
-                if index == -1:
-                    continue
-                cum += state[index]
-        xs[i] += cum
-    if not exclude_association:
-        for i in range(network.n_reac+network.n_prod):
-            xs[i+network.n_isom] += state[-network.n_reac-network.n_prod+i]
-    return xs
+        raise ValueError
+
+    isomers = network.isomers[:]
+    reactants = network.reactants[:]
+    products = network.products[:]
+    xsorder = isomers+reactants+products
+    n_isomreac = len(isomers+reactants)
+    xssource = []
+    dxdtssource = []
+    xseq = []
+    dxdtseq = []
+
+    #Single domainant source simulations
+    for i,isomer in enumerate(isomers):
```

Comment:
Could you add blank after commas in this commit? Also the other ones in this commit

Reply:
done